### PR TITLE
Reorganize dapf

### PR DESCRIPTION
**Pull request by mendess**
- Separate out human friendly parsers of DAP types to seperate module
- Cleanup TestAction::CreateAddTaskJson
- create dap functions module
- Move to functions::helper the code needed to make a request to the helper
    which will allow us to possibly reuse this code in a individual dapf function

### CODE REVIEW

File: crates/dapf/src/cli_parsers.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
+//! Human friendly parsers for common types of parameters to DAP functions.
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (cjpatton) [inline crates/dapf/src/cli_parsers.rs:3]: The copyright notice should be at the top of the file. The module docstring should have newlines around it.

File: crates/dapf/src/cli_parsers.rs [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    vdaf::{Prio3Config, VdafConfig},
+    DapQueryConfig,
+};
+
+#[derive(Debug, Clone, Copy, Default)]
+pub enum DefaultVdafConfigs {
+    Prio2Dimension99k,
+    #[default]
+    Prio3NumProofs2,
+    Prio3NumProofs3,
```
Comment (cjpatton) [inline crates/dapf/src/cli_parsers.rs:26]: These names are very strange and need documentation. For each variant, I would spell out what VDAF is in use and what are its parameters.

I also think it's worth saying something like "these variants are used for load testing".

File: crates/dapf/src/cli_parsers.rs [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                        } else {
+                            default_err
+                        }
+                    })
+            })
+    }
+}
+
+impl CliVdafConfig {
+    pub fn into_vdaf(self) -> VdafConfig {
```
Comment (cjpatton) [inline crates/dapf/src/cli_parsers.rs:118]: "into_config()"?

File: crates/dapf/src/cli_parsers.rs (hunk 2 above)
Comment (cjpatton) [reply crates/dapf/src/cli_parsers.rs:26]: BTW I see this code is just moved, but we might as well add some documentation!

File: crates/daphne-service-utils/src/test_route_types.rs [hunk 4]
```diff
@@ -23,6 +26,47 @@ pub struct InternalTestVdaf {
     pub chunk_length: Option<String>,
 }
 
+impl From<VdafConfig> for InternalTestVdaf {
```
Comment (cjpatton) [inline crates/daphne-service-utils/src/test_route_types.rs:29]: What does "Internal" mean in this context?

File: crates/dapf/src/functions/mod.rs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+//! The various DAP functions dapf can perform.
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (cjpatton) [inline crates/dapf/src/functions/mod.rs:3]: Copyright at the top, module docstring in its own paragraph

File: crates/dapf/src/functions/mod.rs [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+
+pub mod hpke;
+pub mod test_routes;
```
Comment (cjpatton) [inline crates/dapf/src/functions/mod.rs:7]: What does "test routes" mean in this context? If you're referring to the interop API, then document it here. If these includce non-standard APIs that are specific to Daphne, then document this as well.

File: crates/dapf/src/acceptance/mod.rs [hunk 7]
```diff
@@ -509,62 +504,30 @@ impl Test {
...
-                    agg_job_init_req
-                        .get_encoded_with_param(&task_config.version)
-                        .unwrap(),
-                )
-                .headers(headers),
-        )
-        .await?;
+        let agg_job_resp = self
+            .http_client
+            .submit_aggregate_init_req(
```
Comment (cjpatton) [inline crates/dapf/src/acceptance/mod.rs:514]: "aggregation_job_init_req"

**Review by cjpatton — APPROVED** [review]
Minor nits only. Note one of the commit titles is so long that GitHub splits it out into the commit body. I think it's a good idea to make the title fit on one line so that `git log` is easier to read.

File: crates/daphne-service-utils/src/test_route_types.rs (hunk 4 above)
Comment (mendess) [reply crates/daphne-service-utils/src/test_route_types.rs:29]: You came up with this name, [you tell me](https://github.com/cloudflare/daphne/pull/153) :eyes:

File: crates/dapf/src/functions/mod.rs (hunk 6 above)
Comment (mendess) [reply crates/dapf/src/functions/mod.rs:7]: It maps to routes in the file with the same name `daphne-server/src/router/test_routes.rs`. Do you think we should rename all of these to something that references interop more clearly?

File: crates/dapf/src/cli_parsers.rs (hunk 2 above)
Comment (mendess) [reply crates/dapf/src/cli_parsers.rs:27]: I just wanted some quick default vdafs to pass through the commandline. So I wanted the names to be short which means they won't be very good either

File: crates/daphne-service-utils/src/test_route_types.rs (hunk 4 above)
Comment (cjpatton) [reply crates/daphne-service-utils/src/test_route_types.rs:29]: It wasn't clear to me that this was related to one of the /internal/... endpoints. Perhaps just add some documentation.

File: crates/dapf/src/functions/mod.rs (hunk 6 above)
Comment (cjpatton) [reply crates/dapf/src/functions/mod.rs:8]: Thanks for clarifying. No need to rename things, just add some documentation.

File: crates/dapf/src/cli_parsers.rs (hunk 2 above)
Comment (cjpatton) [reply crates/dapf/src/cli_parsers.rs:27]: The names are find, just document :)

**Pull request merged**